weave/docker/Dockerfile.python-client.test

```diff
@@ -10,6 +10,6 @@ RUN python3 -m venv venv
 RUN --mount=type=cache,target=/root/.cache /bin/bash -c \
     "source venv/bin/activate && \
     pip install --upgrade pip && \
-    pip install -r requirements.test.txt"
+    pip install -r requirements.legacy.test.txt"
```

Comment:
This doesn't seem right - the legacy tests should be using `weave/legacy/Dockerfile.test`. This Dockerfile would be used for your tox tests (or deleted if we want to use wandbs)

Reply:
I'm not very familiar here, but I'm following this part of the tests:
https://github.com/wandb/weave/blame/andrew/tox3/.github/workflows/test.yaml#L37-L40

Which seems to be:
1. A unit test image which is used for the legacy tests below
2. An integration test image which is used for the weave tests in core